### PR TITLE
Amazon Q Code Transform: Improve telemetry to capture transform journey

### PR DESCRIPTION
**Pull request by tincheng**
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Description

Emit new metrics to better capture user journey for Amazon Q Code Transform. 
Remove deprecated metrics.

New metrics published:
```
// initiate transform
DefaultMetricEvent(createTime=2024-07-25T23:31:20.681730Z, awsAccount=n/a, awsRegion=n/a, awsProduct=Amazon Q For JetBrains, awsVersion=3.19-SNAPSHOT-233+c0e7ddc.modified, data=[DefaultDatum(name=codeTransform_initiateTransform, value=1.0, unit=None, passive=false, metadata={result=Succeeded, credentialSourceId=iamIdentityCenter, codeTransformSessionId=dd5054d6-35b2-42bd-84fe-8959a57b38c8})])

// validate project
DefaultMetricEvent(createTime=2024-07-25T23:31:22.693252Z, awsAccount=n/a, awsRegion=n/a, awsProduct=Amazon Q For JetBrains, awsVersion=3.19-SNAPSHOT-233+c0e7ddc.modified, data=[DefaultDatum(name=codeTransform_validateProject, value=1.0, unit=None, passive=false, metadata={result=Succeeded, codeTransformLocalJavaVersion=JDK_1_8, buildSystemVersion=Bundled (Maven 3), codeTransformSessionId=dd5054d6-35b2-42bd-84fe-8959a57b38c8, codeTransformBuildSystem=Maven})])

// user submits selection
DefaultMetricEvent(createTime=2024-07-25T23:31:27.925320Z, awsAccount=n/a, awsRegion=n/a, awsProduct=Amazon Q For JetBrains, awsVersion=3.19-SNAPSHOT-233+c0e7ddc.modified, data=[DefaultDatum(name=codeTransform_submitSelection, value=1.0, unit=None, passive=false, metadata={result=Succeeded, userChoice=Confirm, codeTransformJavaSourceVersionsAllowed=JDK_1_8, codeTransformJavaTargetVersionsAllowed=JDK_17, codeTransformProjectId=8sKfQJq2MR72O4GEbFHE8HUAIV/zrqmQRvj/KTCGBtM=, codeTransformSessionId=dd5054d6-35b2-42bd-84fe-8959a57b38c8})])

// build project locally
DefaultMetricEvent(createTime=2024-07-25T23:31:43.864113Z, awsAccount=n/a, awsRegion=n/a, awsProduct=Amazon Q For JetBrains, awsVersion=3.19-SNAPSHOT-233+c0e7ddc.modified, data=[DefaultDatum(name=codeTransform_localBuildProject, value=1.0, unit=None, passive=false, metadata={result=Succeeded, codeTransformBuildCommand=IDEBundledMaven, codeTransformSessionId=dd5054d6-35b2-42bd-84fe-8959a57b38c8})])

// upload project
DefaultMetricEvent(createTime=2024-07-25T23:32:00.554623Z, awsAccount=n/a, awsRegion=n/a, awsProduct=Amazon Q For JetBrains, awsVersion=3.19-SNAPSHOT-233+c0e7ddc.modified, data=[DefaultDatum(name=codeTransform_uploadProject, value=1.0, unit=None, passive=false, metadata={result=Succeeded, codeTransformTotalByteSize=10806548, codeTransformRunTimeLatency=16624, codeTransformDependenciesCopied=true, codeTransformSessionId=dd5054d6-35b2-42bd-84fe-8959a57b38c8})])

// start job
DefaultMetricEvent(createTime=2024-07-25T23:35:46.403011Z, awsAccount=n/a, awsRegion=n/a, awsProduct=Amazon Q For JetBrains, awsVersion=3.19-SNAPSHOT-233+c0e7ddc.modified, data=[DefaultDatum(name=codeTransform_jobStart, value=1.0, unit=None, passive=false, metadata={result=Succeeded, codeTransformRunTimeLatency=18336, codeTransformJobId=a19bf628-31df-4085-ae4c-4a736da7ae13, codeTransformSessionId=f37233f4-74fe-4e01-8c2d-f91315d95d0c})])

// download result
DefaultMetricEvent(createTime=2024-07-25T23:45:01.364038Z, awsAccount=n/a, awsRegion=n/a, awsProduct=Amazon Q For JetBrains, awsVersion=3.19-SNAPSHOT-233+c0e7ddc.modified, data=[DefaultDatum(name=codeTransform_downloadArtifact, value=1.0, unit=None, passive=false, metadata={result=Succeeded, codeTransformTotalByteSize=360556, codeTransformRunTimeLatency=3391, codeTransformJobId=9688af04-e36a-41e8-b980-0e5298ebb7d6, codeTransformSessionId=fb8d0a93-4981-4baa-b721-bc683b2a8be1, codeTransformArtifactType=ClientInstructions})])

// view result
DefaultMetricEvent(createTime=2024-07-25T23:45:13.256535Z, awsAccount=n/a, awsRegion=n/a, awsProduct=Amazon Q For JetBrains, awsVersion=3.19-SNAPSHOT-233+c0e7ddc.modified, data=[DefaultDatum(name=codeTransform_viewArtifact, value=1.0, unit=None, passive=false, metadata={result=Succeeded, userChoice=Cancel, codeTransformStatus=COMPLETED, codeTransformJobId=9688af04-e36a-41e8-b980-0e5298ebb7d6, codeTransformSessionId=fb8d0a93-4981-4baa-b721-bc683b2a8be1, codeTransformVCSViewerSrcComponents=toastNotification, codeTransformArtifactType=ClientInstructions})])
```

## Checklist
- [x] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [ ] A short description of the change has been added to the **[CHANGELOG](https://github.com/aws/aws-toolkit-jetbrains/blob/master/CONTRIBUTING.md#contributing-via-pull-requests)** if the change is customer-facing in the IDE.
- [x] I have added metrics for my changes (if required)
 
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: plugins/amazonq/codetransform/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codemodernizer/ideMaven/MavenRunnerUtils.kt [hunk 1]
```diff
@@ -49,11 +44,8 @@ fun runHilMavenCopyDependency(
             buildlogBuilder.appendLine(successMsg)
         } else if (copyDependenciesRunnable.isTerminated()) {
             return MavenCopyCommandsResult.Cancelled
-        } else {
-            emitMavenFailure("Maven Copy: bundled Maven failed: exitCode ${copyDependenciesRunnable.isComplete()}", logger, telemetry)
```
Comment (dhasani23) [inline plugins/amazonq/codetransform/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codemodernizer/ideMaven/MavenRunnerUtils.kt:53]: Do we have another way of knowing that the local build failed?

Comment (dhasani23) [reply plugins/amazonq/codetransform/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codemodernizer/ideMaven/MavenRunnerUtils.kt:53]: Oh whoops I see it below now, nevermind.

File: plugins/amazonq/codetransform/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codemodernizer/CodeModernizerManager.kt [hunk 2]
```diff
@@ -172,22 +175,32 @@ class CodeModernizerManager(private val project: Project) : PersistentStateCompo
...
+                    invalidReason = message(
+                        "codemodernizer.notification.warn.invalid_project.description.reason.no_valid_files",
+                        supportedBuildFileNames.joinToString()
+                    ),
+                    invalidTelemetryReason = InvalidTelemetryReason(
                         CodeTransformPreValidationError.NonMavenProject,
                         if (isGradleProject(project)) "Gradle build" else "other build"
-                    )
+                    ),
+                    buildSystem = if (isGradleProject(project)) CodeTransformBuildSystem.Gradle else CodeTransformBuildSystem.Unknown
```
Comment (dhasani23) [inline plugins/amazonq/codetransform/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codemodernizer/CodeModernizerManager.kt:196]: Right now this `isGradleProject()` is just used for telemetry right? Looks like it's implemented here: https://github.com/aws/aws-toolkit-jetbrains/blob/3c8beb6ddf411daf1131d9525733c644a95985c3/plugins/amazonq/codetransform/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codemodernizer/utils/CodeTransformValidationUtils.kt#L10

**Review by dhasani23 — APPROVED** [review]

File: plugins/amazonq/codetransform/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codemodernizer/CodeModernizerManager.kt (hunk 2 above)
Comment (tincheng) [reply plugins/amazonq/codetransform/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codemodernizer/CodeModernizerManager.kt:196]: yes! It's just for telemetry and it's using the function from the util.

**Comment by tincheng** [comment]
Instead of removing the deprecated metrics, I should keep them for short term backward compatibility for BI analysis purpose. I will add them back, and remove them after BI team confirms no longer needing the old metrics.

**Review by rli — APPROVED** [review]

**Pull request merged**